hera_sim/visibilities/vis_cpu.py

```diff
@@ -318,33 +405,51 @@ def vis_cpu(antpos, freq, eq2tops, crd_eq, I_sky, bm_cube,
     tau = np.empty((nant, npix), dtype=real_dtype)
     v = np.empty((nant, npix), dtype=complex_dtype)
     crd_eq = crd_eq.astype(real_dtype)
-
-    bm_pix_x = np.linspace(-1, 1, bm_pix)
-    bm_pix_y = np.linspace(-1, 1, bm_pix)
-
-    # Loop over time samples.
-    for t, eq2top in enumerate(eq2tops.astype(real_dtype)):
-        tx, ty, tz = crd_top = np.dot(eq2top, crd_eq)
-
+    
+    # Precompute splines is using pixelized beams
+    if beam_list is None:
+        bm_pix_x = np.linspace(-1, 1, bm_pix)
+        bm_pix_y = np.linspace(-1, 1, bm_pix)
+    
+        splines = []
         for i in range(nant):
             # Linear interpolation of primary beam pattern.
-            spline = RectBivariateSpline(bm_pix_y, bm_pix_x, bm_cube[i], kx=1,
-                                         ky=1)
-            A_s[i] = spline(ty, tx, grid=False)
+            spl = RectBivariateSpline(bm_pix_y, bm_pix_x, bm_cube[i], kx=1, ky=1)
+            splines.append(spl)
+    
+    # Loop over time samples
+    for t, eq2top in enumerate(eq2tops.astype(real_dtype)):
+        tx, ty, tz = crd_top = np.dot(eq2top, crd_eq)
+        
+        # Primary beam response
+        if beam_list is None:
+            # Primary beam pattern using pixelized primary beam
+            for i in range(nant):
+                A_s[i] = splines[i](ty, tx, grid=False)
+                # FIXME: Try using a log-space beam for accuracy!
+        else:
+            # Primary beam pattern using direct interpolation of UVBeam object
+            az, za = conversions.lm_to_az_za(ty, tx) # FIXME: Order of tx, ty
```

Comment:
has the order of tx, ty been checked?

Reply:
Also need to check conversions.lm_to_az_za. What we want is that the pixel_beam method and the direct interpolation method give roughly the same beam value around an asymmetrical beam. Then they are consistent within hera_sim. I will check that.

Reply:
I checked the values returned by the direct interpolation against those returned by the pixel beam interpolation, using several source orientations. This shows that the order should be tx, ty in lm_to_az_za(). This will be checked in the test script. It is changed.